src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,84 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+
+    public static final String OPERATION_VALUE_DELIMITER = ",";
+    public static final String SUPPLY_OPERATION_TITLE = "supply";
+    public static final String BUY_OPERATION_TITLE = "buy";
+    public static final String RESULT_OPERATION_TITLE = "result";
+    public static final int OPERATION_INDEX = 0;
+    public static final int VALUE_INDEX = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        writeFile(composeReport(calculateOperations(readFile(fromFileName))), toFileName);
+    }
+
+    public String[] readFile(String filename) {
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(filename))) {
```

Comment:
Good use of try-with-resources to ensure the `BufferedReader` is closed properly.